### ISSUE TITLE
Report an error in python3.8：WinError 10022

### ISSUE THREAD

**Issue by cafkagh** [kyan001/ping3#28]
File "D:/Code/Mercury/Mercury/qt/test.py", line 164, in <module>
    ping3.ping("www.microsoft.com")
  File "D:\Code\Mercury\Mercury\qt\venv\lib\site-packages\ping3.py", line 82, in wrapper
    func_return = func(*args, **kwargs)
  File "D:\Code\Mercury\Mercury\qt\venv\lib\site-packages\ping3.py", line 280, in ping
    sock.setsockopt(socket.SOL_IP, socket.IP_TTL, ttl)
OSError: [WinError 10022] 提供了一个无效的参数。

**Comment by kyan001** [comment]
Hi @cafkagh ,

I just tested on my win10 with python3.8.3, there is no such error. Can you provide more info for debugging? Such as your Windows version (mine is Win10 1803), your python3 version such as "3.8.?", and the code you used?

Thanks for the feedback, I will look into it.
```py
Python 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import ping3
>>> ping3.ping("www.microsoft.com")
0.013941764831542969
>>>
```

**Comment by kyan001** [comment]
I checked [this microsoft docs page](https://docs.microsoft.com/en-us/windows/win32/winsock/ipproto-ip-socket-options) and found:

> IP_TTL: 
>
> Changes the default value set by the TCP/IP service provider in the TTL field of the IP header in outgoing datagrams. IP_TTL support is not required; to check whether IP_TTL is supported, use getsockopt to get current options. If getsockopt fails, IP_TTL is not supported.

so I guess I could check the IP_TTL availability by using following code:
```py
with socket.socket(socket.AF_INET, socket.SOCK_RAW, socket.IPPROTO_ICMP) as sock:
     print(sock.setsockopt(socket.SOL_IP, socket.IP_TTL))
```
Can you run this code as admin and tell me the result?

**Comment by nestlin** [comment]
I also encounter this issue with win8.1 + python3.8.5

![image](https://user-images.githubusercontent.com/62738491/89258312-7e255d00-d65a-11ea-9a42-eeb0b0de3b4f.png)

**Comment by kyan001** [comment]
> I also encounter this issue with win8.1 + python3.8.5.
> 
> 
> 
> Below is the output by running above code.
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/62738491/89258312-7e255d00-d65a-11ea-9a42-eeb0b0de3b4f.png)
> 
> 

Great help. thanks I'll look into it.

**Comment by nestlin** [comment]
This can work with IPPROTO_IP.

![image](https://user-images.githubusercontent.com/62738491/89259423-dbbaa900-d65c-11ea-86e1-b361399caf20.png)

**Comment by kyan001** [comment]
> 
> 
> This can work with IPPROTO_IP.
> 
> ![image](https://user-images.githubusercontent.com/62738491/89259423-dbbaa900-d65c-11ea-86e1-b361399caf20.png)

Thanks @nestlin 

I fixed this in 2.6.6, which try to set both IPPROTO_IP and SOL_IP.
